### PR TITLE
Created Base File

### PR DESCRIPTION
**Pull request by RafaBonin**
Created a Base File for connecting, receiving data, parsing data and sending data

### CODE REVIEW

File: netmosquito.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import socket
+
+host='127.0.0.1' #Host IP goes here
```
Comment (olivatooo) [inline netmosquito.py]: Maybe we can receive this with arguments?

File: netmosquito.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+port=7000		 #Port Goes Here
+
+s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+s.connect(host,port)
+
+def recieve_data(s):
```
Comment (olivatooo) [inline netmosquito.py]: This function does not receive any data < 2048. This needs a better receive function

Comment (olivatooo) [reply netmosquito.py]: mispelled `receive`

File: netmosquito.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	data = s.recv(2048)
+	return data
+
+def parse_data(raw_data):
```
Comment (olivatooo) [inline netmosquito.py:10]: This always returns None

File: netmosquito.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	return response
+
+def send_data(parsed_data):
```
Comment (olivatooo) [inline netmosquito.py]: `s` is global, maybe specify this, or send socket as a param

File: netmosquito.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+	s.send(parsed_data)
+
+while True:
```
Comment (olivatooo) [inline netmosquito.py:23]: We need a better condition! While True is not good at all

File: netmosquito.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+	raw_data = recieve_data(s)
+	parsed_data = parsed_data(raw_data)
+	send_data(parsed_data)
```
Comment (olivatooo) [inline netmosquito.py]: Missing blank line at the end of file

**Review by olivatooo — CHANGES_REQUESTED** [review]

File: netmosquito.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import socket, sys
+
+try:
```
Comment (olivatooo) [inline netmosquito.py]: Don't make this check with try catch, and display a usage of the program with an example

File: netmosquito.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+	exit()
+
+s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+s.connect((host,port))
+
+def receive_data(s):
+	#This functions recieves data in 2048bytes chunks until '\n' is received
+	data=''
+	while True:
+		data+=s.recv(2048).decode()
```
Comment (olivatooo) [inline netmosquito.py]: Too big, put 4 bytes or less!

File: netmosquito.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+	print("Please inform host and port uppon executing")
+	exit()
+
+s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+s.connect((host,port))
+
+def receive_data(s):
+	#This functions recieves data in 2048bytes chunks until '\n' is received
+	data=''
+	while True:
```
Comment (olivatooo) [inline netmosquito.py]: fix this condition

**Review by olivatooo — CHANGES_REQUESTED** [review]

File: netmosquito.py [hunk 10]
```diff
@@ -0,0 +1,68 @@
+import socket, sys
+
+buffer_size = 1				#How much data would you like to receive at once (bytes)
```
Comment (olivatooo) [inline netmosquito.py]: Unaligned commentary

File: netmosquito.py [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+timeout_time= 2				#Timeout to not keep holding forever in case of incomplete packet
```
Comment (olivatooo) [inline netmosquito.py]: Unaligned commentary

File: netmosquito.py [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+if len(sys.argv)==3:
+	host=sys.argv[1] 		#Host IP goes here
```
Comment (olivatooo) [inline netmosquito.py]: Unaligned commentary

File: netmosquito.py [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+	port=int(sys.argv[2]) 	#Port Goes Here
```
Comment (olivatooo) [inline netmosquito.py]: Unaligned commentary

File: netmosquito.py [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+
+else:
+	print(('Error while running, please execute proggram as following:\npython netmosquito.py <host> <port>'))
```
Comment (olivatooo) [inline netmosquito.py]: Double `( )`

File: netmosquito.py [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+	exit()
+
+s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+s.connect((host,port))
+print(f'Connection established with {host}:{str(port)}\n')
+s.settimeout(timeout_time)	#Sets the timeout time
```
Comment (olivatooo) [inline netmosquito.py]: Unaligned commentary

File: netmosquito.py [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+
+def receive_data(s):
+	#This functions recieves data in n bytes chunks until '\n' is received or timeout ocurrs
```
Comment (olivatooo) [inline netmosquito.py]: Commentary in function without [doc string](https://www.askpython.com/python/python-comments)

Comment (olivatooo) [reply netmosquito.py]: Unaligned commentary

File: netmosquito.py [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+	data=''
+	try:					#Prevents exception from timeouts
```
Comment (olivatooo) [inline netmosquito.py]: Unaligned commentary

File: netmosquito.py [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+		data+=s.recv(buffer_size).decode()
+	except socket.timeout:
+		return data
+
+	while data[-1]!='\n':
+		try:					#Prevents exception from timeouts
```
Comment (olivatooo) [inline netmosquito.py]: Unaligned commentary

File: netmosquito.py [hunk 19]
```diff
@@ -0,0 +1,68 @@
+import socket, sys
+
+buffer_size = 1				#How much data would you like to receive at once (bytes)
+timeout_time= 2				#Timeout to not keep holding forever in case of incomplete packet
+
+if len(sys.argv)==3:
```
Comment (olivatooo) [inline netmosquito.py]: Put this initial check into a function

**Review by olivatooo — CHANGES_REQUESTED** [review]
Apply pep-8 to file, separate all socket functions into another file and only import here

**Review by olivatooo — APPROVED** [review]
mano, dá play, se der ruim amanhã já sabe que o piriquito vai cantar

**Review by sugayaa — APPROVED** [review]

**Pull request merged**